request/extractor.go

```diff
@@ -79,3 +80,16 @@ func (e *PostExtractionFilter) ExtractToken(req *http.Request) (string, error) {
 		return "", err
 	}
 }
+
+// BearerExtractor extracts a token from the Authorization header.
+// The header is expected to match the format "Bearer XX", where "XX" is the
+// JWT token.
+type BearerExtractor struct{}
+
+func (e BearerExtractor) ExtractToken(req *http.Request) (string, error) {
+	tokenHeader := req.Header.Get("Authorization")
+	if tokenHeader == "" || !strings.HasPrefix(tokenHeader, "Bearer ") {
```

Comment:
You might consider making this case insensitive. The headers are definitely case insensitive, but Headers.Get already takes care of this. I need to Look it up again but i am pretty sure `bearer` is also ok.

Reply:
I tried to find the actual RFC that specifies this, which is... let's say complicated. There are some who think that it is insensitive (see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization#directives). However, it is not really 100 % specified in the RFC. It is only specified in the `WWW-Authenticate` server response (to be case-insensitive), so I would keep this as is for now.

Unless some RFC expert can shed some more insight on this.

Reply:
Updates this to be case-insensitive. It's best to be flexible here.

Reply:
Also tests cover this now.